### PR TITLE
[Storage] Add new `copy_source_tags` option to `start_copy_from_url()`

### PR DESCRIPTION
**Pull request by jalauzon-msft**
STG82 adds a new option for synchronous copy operations that is called `copy_source_tags`. This option allows callers to specify whether to copy or replace source blob tags during a copy operation.

NOTE: The generated code for this change includes code for other STG82 features as well, which will be implemented in other PRs.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py [hunk 1]
```diff
@@ -3628,6 +3632,9 @@ def copy_from_url(  # pylint: disable=inconsistent-return-statements
         :param copy_source_authorization: Only Bearer type is supported. Credentials should be a valid
          OAuth access token to copy source. Default value is None.
         :type copy_source_authorization: str
+        :param copy_source_tags: Optional, default 'replace'.  Indicates if source tags should be
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py:3635]: nit: Remove one of the whitespaces between the period and the capital I in 'Indicates'

File: sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_block_blob_operations.py [hunk 2]
```diff
@@ -889,6 +893,9 @@ def put_blob_from_url(  # pylint: disable=inconsistent-return-statements
         :param copy_source_authorization: Only Bearer type is supported. Credentials should be a valid
          OAuth access token to copy source. Default value is None.
         :type copy_source_authorization: str
+        :param copy_source_tags: Optional, default 'replace'.  Indicates if source tags should be
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_block_blob_operations.py:896]: Same nit: Extra whitespace

**Review by vincenttran-msft — CHANGES_REQUESTED** [review]
LGTM! Just resolve the extra whitespace and should be good to go :)

File: sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py (hunk 1 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py:3635]: This is generated code so unfortunately, I cannot change it :)

File: sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_block_blob_operations.py (hunk 2 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_block_blob_operations.py:896]: Same comment as above, this is generated.

**Pull request merged**